pkg/controller/controller.go

```diff
@@ -265,6 +270,39 @@ func (c *Controller) scaleNodeGroup(nodegroup string, nodeGroup *NodeGroupState)
 		return nodeGroup.scaleUpLock.requestedNodes, nil
 	}
 
+	// If we are not locked, we're either init or after a scale event
```

Comment:
Are you able to move this block of code into a seperate method? The current method it's in is getting quite large